packages/compass-components/src/components/interactive-popover.tsx

```diff
@@ -50,6 +50,8 @@ type InteractivePopoverProps = {
     ref: React.LegacyRef<HTMLButtonElement>;
     children: React.ReactNode;
   }) => React.ReactElement;
+  hasCustomCloseButton?: boolean;
```

Comment:
Can we come up with a better name here? If something has a button, you'd expect it to show and not be hidden?

Reply:
Using custom, for specifying custom buttons, is something we use in leafy-green and in the code-base already:

https://www.mongodb.design/component/code/documentation/

And also for additional business logic, like in CodemirrorMultilineEditor, where we have customActions.

(I know these examples because are the ones I used as inspiration).

It could be an array of buttons like in the Code component, but it's more complex than an optional boolean for this case, where we just want to have a custom action for closing.

Reply:
That's exactly what I'm saying: both `customActions` and `customActionButtons` **_adds_** additional elements that render inside the component when provided, you added a property called `hasCustomCloseButton` that **_removes_** the element rendered inside the component when provided

Reply:
I could change the name to overrideCloseButton, is it better?

Reply:
I'd go with hideCloseButton as this is literally what this property does, you're not passing any overrides here, nor you can force this somehow with this property, but feel free to pick up whatever makes sense to you or just resolve this

Reply:
I'll change it to the name you suggested and resolve.